test/RpcEvmCompatibility.js

```diff
@@ -0,0 +1,298 @@
+    });
+
+    it('should retrieve the code of a contract', async () => {
+        // Make the eth_getCode call for the deployed contract
+        const code = await provider.send('eth_getCode', [rpcCompatibilityContract.address, 'latest']);
+
+        // Verify the code
+        expect(code).to.be.a('string');
+        expect(/^0x[0-9a-fA-F]*$/.test(code)).to.be.true; // Ensure it's a valid hexadecimal string
+        expect(code).to.not.equal('0x'); // Ensure it's not an empty code
+    });
+
+    it('should estimate gas for a transaction', async () => {
+        const gasLimit = network.config.gasOptions?.gasLimit || 50000;
+        const newValue = 100;
+        const transactionParams = {
+            to: rpcCompatibilityContract.address,
+            data: rpcCompatibilityContract.interface.encodeFunctionData('updateValue', [newValue]),
+            gasLimit,
+        };
+
+        // Make the eth_estimateGas call
+        const estimatedGas = await provider.send('eth_estimateGas', [transactionParams]);
+
+        // Verify the estimated gas
+        expect(estimatedGas).to.be.a('string');
+        expect(ethers.BigNumber.from(estimatedGas).gt(0)).to.be.true;
+    });
+
+    it('should retrieve the current gas price', async () => {
+        // Make the eth_gasPrice call
+        const gasPrice = await provider.send('eth_gasPrice', []);
+
+        // Verify the gas price
+        expect(gasPrice).to.be.a('string');
+        expect(ethers.BigNumber.from(gasPrice).toNumber()).to.be.above(0); // Gas price should be a positive integer
+    });
+
+    it('should retrieve the chain ID', async () => {
+        // Make the eth_chainId call
+        const chainId = await provider.send('eth_chainId', []);
+
+        // Verify the chain ID
+        expect(chainId).to.be.a('string');
+        expect(ethers.BigNumber.from(chainId).toNumber()).to.be.above(0); // Chain ID should be a positive integer
+    });
+
+    it('should retrieve the transaction count of an address', async () => {
+        // Make the eth_getTransactionCount call
+        const transactionCount = await provider.send('eth_getTransactionCount', [signers[0].address, 'latest']);
+
+        // Verify the transaction count
+        expect(transactionCount).to.be.a('string');
+        expect(ethers.BigNumber.from(transactionCount).toNumber()).to.be.at.least(0); // Transaction count should be a non-negative integer
+    });
+
+    it('should send a raw transaction', async () => {
+        const privateKeys = isHardhat ? KnownAccount0PrivateKeyHardhat : keys?.accounts || keys.chains[network.name]?.accounts;
+        const wallet = new Wallet(privateKeys[0], provider);
+
+        let tx = {
+            to: rpcCompatibilityContract.address,
+            data: rpcCompatibilityContract.interface.encodeFunctionData('getValue'),
+            gasLimit: network.config.gasOptions?.gasLimit || 23310, // Use an appropriate gas limit
+        };
+
+        tx = await signer.populateTransaction(tx);
```

Comment:
```suggestion
        tx = await signer.populateTransaction(rpcCompatibilityContract.populateTransaction.updateValue(newValue));
```
no need to build the tx object manually